### PR TITLE
MAINT: remove python 3.5 from builds for 1.19, as per NEP 29

### PR DESCRIPTION
**Pull request by mattip**
remove python 3.5 as a supported platform, as per NEP 29

### CODE REVIEW

**Comment by charris** [comment]
If we do this, it definitely needs a release note because NumPy won't work with Python 3.5.

**Comment by mattip** [comment]
updated as per mailing list feedback to only remove 3.5 from the builds and PyPI spec

File: setup.py [hunk 1]
```diff
@@ -27,8 +27,8 @@
 import textwrap
 
 
-if sys.version_info[:2] < (3, 5):
-    raise RuntimeError("Python version >= 3.5 required.")
+if sys.version_info[:2] < (3, 6):
+    raise RuntimeError("Python version >= 3.6 required.")
```
Comment (mattip) [inline setup.py]: I guess this should be reverted too

**Comment by rgommers** [comment]
As commented on the mailing list, let's either drop it completely, or keep it supported for one more month.

**Comment by rgommers** [comment]
To be merged only for 1.19, see https://mail.python.org/pipermail/numpy-discussion/2019-October/080191.html

**Comment by mattip** [comment]
Closing. This was already done for 1.19

**Pull request closed without merging**